app/screens/image_preview/previewer.js

```diff
@@ -0,0 +1,277 @@
+            imageHeight: new Animated.Value(props.imageHeight),
+            imageWidth: new Animated.Value(props.imageWidth),
+            opacity: new Animated.Value(0),
+            requesting: true,
+            retry: 0,
+            zooming: false
+        };
+    }
+
+    componentWillMount() {
+        this.panResponder = PanResponder.create({
+            onStartShouldSetPanResponderCapture: (evt, gestureState) => {
+                const {numberActiveTouches} = gestureState;
+                if (numberActiveTouches === 1 && !this.state.isZooming) {
+                    return true;
+                } else if (numberActiveTouches === 1) {
+                    this.handleResponderRelease(evt);
+                }
+
+                return false;
+            },
+            onPanResponderGrant: () => {
+                return;
+            },
+            onPanResponderTerminate: () => {
+                return;
+            },
+            onShouldBlockNativeResponder: () => false
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.shrink && !nextProps.shrink) {
+            this.setShrink();
+        } else if (!this.props.shrink && nextProps.shrink) {
+            this.setShrink(true);
+        }
+    }
+
+    setShrink = (shrink = false) => {
+        const {gutter, imageHeight, imageWidth} = this.props;
+
+        let height = imageHeight;
+        let width = imageWidth;
+        const duration = 150;
+        if (shrink) {
+            height = height - gutter;
+            width = width - gutter;
+        }
+
+        const animations = [
+            Animated.timing(this.state.imageWidth, {
+                toValue: width,
+                duration
+            })
+        ];
+
+        if (Platform.OS === 'android') {
+            animations.push(
+                Animated.timing(this.state.imageHeight, {
+                    toValue: height,
+                    duration
+                })
+            );
+        }
+
+        Animated.parallel(animations).start();
+    }
+
+    handleResponderRelease = (evt) => {
+        clearTimeout(this.singleTap);
+        let cancelSingleTap = false;
+        if (this.lastTap && Date.now() - this.lastTap < DOUBLE_CLICK_THRESHOLD) {
+            cancelSingleTap = true;
+        } else {
+            this.lastTap = Date.now();
+        }
+
+        if (cancelSingleTap) {
+            const {nativeEvent} = evt;
+            const x = nativeEvent.locationX;
+            const y = nativeEvent.locationY;
+
+            cancelSingleTap = false;
+            this.lastTap = null;
+
+            this.props.onImageDoubleTap(x, y);
+        } else if (!this.state.isZooming) {
+            this.singleTap = setTimeout(() => {
+                this.props.onImageTap();
+            }, DOUBLE_CLICK_THRESHOLD);
+        }
+    }
+
+    handleLoadError = () => {
+        if (this.state.retry < 4) {
+            setTimeout(() => {
+                this.setState({
```

Comment:
Also, since this is delayed, we should use alternate version of setState that will update based on the state after the timeout instead of the state when the timeout is set
```
this.setState((prevState) => {
    return {
        retry: prevState.retry + 1,
        timestamp: Date.now()
    };
});
```

Right now, if `handleLoadError` is called twice in quick succession for whatever reason, you'll have one call pave over the results of the other.